cpp/perspective/src/cpp/emscripten.cpp

```diff
@@ -71,8 +81,27 @@ namespace binding {
         } else {
             return false;
         }
-        tm local_tm = *localtime(&tt);
-        (*out) = t_date(local_tm.tm_year + 1900, local_tm.tm_mon, local_tm.tm_mday);
+
+        // TODO: Perspective stores date/datetime in UTC, so if we use
+        // `std::localtime` here, the value will be localized *twice*: once
+        // here and once when the date is exported. I think this is what is
+        // happening; changing `localtime` to `gmtime` fixed date filtering
+        // so that typing in "03/01/2020" actually filters down to "03/01/2020"
+        // and not "02/29/2020" because the value in the table was converted
+        // localized to 02/29/2020 7pm EST from 03/01/2020 12AM UTC.
+        //
+        // what i am not certain about - the previous implementation of this
+        // function in `jsdate_to_t_date` gets the ymd in local time as well
+        // straight from the date object. HOWEVER this issue was present in
+        // 0.4.8 - which leads me to believe that this has been an issue all
+        // along.
+        //
+        // we should maybe have a separate node test process that runs with
+        // TZ="US-Eastern" or something like that to assert the timezone
+        // differences.
+        // ...I think...
+        tm utc_tm = *std::gmtime(&tt);
```

Comment:
@texodus this change fixed date filtering from:

![Screen Shot 2021-01-05 at 9 49 38 PM](https://user-images.githubusercontent.com/13220267/103723787-d7deda00-4fa0-11eb-9fd7-b6c3a9fd42fa.png)

to:

![Screen Shot 2021-01-05 at 9 47 23 PM](https://user-images.githubusercontent.com/13220267/103723791-d8777080-4fa0-11eb-95ef-bc8f1e7d8108.png)

I think this issue was around for a while (earliest version I tested was 0.4.8, far before the date parser rewrite). What I *think* is happening - `Date.getTime()` always uses UTC, so the timestamps stored by Perspective are in UTC, and then re-localized to local time when they are passed into `new Date()` when output. However, this function (and its previous implementation, `jsdate_to_t_date`) pulled the ymd values as local time, which then causes a re-localization on top of an already localized datetime - thus the issue happening in the screenshot above. I would like to verify this with a proper test suite akin to what we have in the Python tests - that will be coming as part of this patch.

What I am unsure about - if datetime values are stored as UTC timestamps and correctly localized, then why does `==` on a datetime still show an offset:

![Screen Shot 2021-01-05 at 10 00 10 PM](https://user-images.githubusercontent.com/13220267/103724189-c0542100-4fa1-11eb-99c5-d8acd7c0b8d0.png)

This behavior has also been around since 0.4.8.